### PR TITLE
feat: confluence frontend Plugin

### PR DESCRIPTION
**Pull request by Prateek-Wayne**
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)

### CODE REVIEW

File: workspaces/confluence/.yarnrc.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+compressionLevel: mixed
```
Comment (ScottGuymer) [inline workspaces/confluence/.yarnrc.yml:1]: What purpose does this have?

File: workspaces/confluence/plugins/search-confluence-backend/api-report.md.api.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+## API Report File for "@philips-software/backstage-plugin-search-confluence-backend"
```
Comment (ScottGuymer) [inline workspaces/confluence/plugins/search-confluence-backend/api-report.md.api.md:1]: Why is this added here? Does not seem you have changed the rest of the package.

File: workspaces/confluence/plugins/search-confluence-frontend/README.md [hunk 3]
```diff
@@ -1,13 +1,18 @@
 # search-confluence-frontend
```
Comment (ScottGuymer) [inline workspaces/confluence/plugins/search-confluence-frontend/README.md:1]: This readme needs to give more information about what this plugin does and specific details on how to install and configure it

File: workspaces/confluence/.yarnrc.yml (hunk 1 above)
Comment (Prateek-Wayne) [reply workspaces/confluence/.yarnrc.yml]: The `.yarnrc.yml` file with the`compressionLevel`: `mixed setting` was automatically generated when I was using Yarn 4. Since I have now switched back to Yarn 1.22, this file is no longer needed and can be removed.
Thanks for pointing out

File: workspaces/confluence/plugins/search-confluence-frontend/README.md (hunk 3 above)
Comment (Prateek-Wayne) [reply workspaces/confluence/plugins/search-confluence-frontend/README.md]: I have updated the README to include more detailed information about the plugin.

File: workspaces/confluence/plugins/search-confluence-backend/api-report.md.api.md (hunk 2 above)
Comment (Prateek-Wayne) [reply workspaces/confluence/plugins/search-confluence-backend/api-report.md.api.md]: it appears as changed,but it has 0 diffs

File: workspaces/confluence/.yarnrc.yml (hunk 1 above)
Comment (ScottGuymer) [reply workspaces/confluence/.yarnrc.yml]: You should be using yarn 4+ why did you move back?

File: workspaces/confluence/plugins/search-confluence-backend/api-report.md.api.md (hunk 2 above)
Comment (ScottGuymer) [reply workspaces/confluence/plugins/search-confluence-backend/api-report.md.api.md]: you should make sure you have correctly rebased then

File: workspaces/confluence/.yarnrc.yml (hunk 1 above)
Comment (Prateek-Wayne) [reply workspaces/confluence/.yarnrc.yml]: I'm not entirely sure. On my laptop, Yarn 4 defaults to Plug'n'Play, and I had to modify the .yarnrc.yml file to use node_modules. This extra step led me to switch back to Yarn 1.22 for simplicity.

I would appreciate your feedback on what I should use in the future. Thanks!

Comment (ScottGuymer) [reply workspaces/confluence/.yarnrc.yml]: yeah the default is pnp but you should just add the yarnrc to make it node modules.

Yarn 4 is _much_ faster and improved, even just using node_modules. no reason not to use it

Comment (Prateek-Wayne) [reply workspaces/confluence/.yarnrc.yml]: I have added .yarnrc file with `nodeLinker: node-modules`

**Review by ScottGuymer — APPROVED** [review]
